src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -277,6 +285,117 @@ class F extends ValidationTest {
 
     validateFinish(success);
   }
+
+  runDefaultLayoutBindingTest<T extends GPURenderPipeline | GPUComputePipeline>({
+    visibility,
+    empty,
+    pipelineType,
+    bindingType,
+    makePipelinesFn,
+    doCommandFn,
+  }: {
+    visibility: number;
+    empty: boolean;
+    pipelineType: PipelineType;
+    bindingType: BindingType;
+    makePipelinesFn: (t: F, explicitPipelineLayout: GPUPipelineLayout) => T[];
+    doCommandFn: (params: {
+      t: F;
+      encoder: GPUCommandEncoder;
+      pipeline: T;
+      emptyBindGroup: GPUBindGroup;
+      nonEmptyBindGroup: GPUBindGroup;
+    }) => void;
+  }) {
+    const { device } = this;
+    const explicitEmptyBindGroupLayout = device.createBindGroupLayout({
+      entries: [],
+    });
+    const explicitBindGroupLayout = device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility,
+          buffer: {},
+        },
+      ],
+    });
+    const explicitPipelineLayout = device.createPipelineLayout({
+      bindGroupLayouts: [explicitEmptyBindGroupLayout, explicitBindGroupLayout],
+    });
+
+    const [pipelineAuto0, pipelineAuto1, pipelineExplicit] = makePipelinesFn(
+      this,
+      explicitPipelineLayout
+    );
+
+    const buffer = device.createBuffer({
+      size: 16,
+      usage: GPUBufferUsage.UNIFORM,
+    });
+    this.trackForCleanup(buffer);
+
+    const kEmptyBindGroupNdx = 0;
+    const kNonEmptyBindGroupNdx = 1;
+
+    const {
+      emptyBindGroupLayout,
+      nonEmptyBindGroupLayout,
+      pipeline,
+    }: {
+      emptyBindGroupLayout: GPUBindGroupLayout;
+      nonEmptyBindGroupLayout: GPUBindGroupLayout;
+      pipeline: T;
+    } = (() => {
+      if (empty) {
+        // Testing empty so
+        //   nonEmptyBindGroupLayout comes from the pipeline we'll render/compute with
+        //   emptyBindGroupLayout comes from a possibly incompatible place.
```

Comment:
nit: to be able to read where the sentence breaks are
```suggestion
        // Testing empty:
        // - nonEmptyBindGroupLayout comes from the pipeline we'll render/compute with.
        // - emptyBindGroupLayout comes from a possibly incompatible place.
```